### PR TITLE
prevents export with no submissions

### PR DESCRIPTION
**Pull request by jasonchung1871**
# Description
If a reviewer is trying to export submissions with no submissions, it gives an infinite spinner. This will return an error to the user indicating that there are no submissions if there are no submissions. A valid submission is a form that is not in the draft state or deleted state.

resolves: https://bcdevex.atlassian.net/browse/FORMS-857

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/package-lock.json [hunk 1]
```diff
@@ -1542,9 +1542,9 @@
       }
     },
     "node_modules/@types/express-serve-static-core": {
-      "version": "4.17.41",
-      "resolved": "https://registry.npmjs.org/@types/express-serve-static-core/-/express-serve-static-core-4.17.41.tgz",
-      "integrity": "sha512-OaJ7XLaelTgrvlZD8/aa0vvvxZdUmlCn6MtWeB7TkiKW70BQLc9XEPpDLPdbo52ZhXUCrznlWdCHWxJWtdyajA==",
+      "version": "4.17.42",
+      "resolved": "https://registry.npmjs.org/@types/express-serve-static-core/-/express-serve-static-core-4.17.42.tgz",
+      "integrity": "sha512-ckM3jm2bf/MfB3+spLPWYPUH573plBFwpOhqQ2WottxYV85j1HQFlxmnTq57X1yHY9awZPig06hL/cLMgNWHIQ==",
```
Comment (WalterMoar) [inline app/package-lock.json:1547]: Is this file needed for the change? It's OK if it gets included in the PR, as we need to update the dependencies, but just curious.

Comment (jasonchung1871) [reply app/package-lock.json:1547]: It is not required!

**Comment by WalterMoar** [comment]
Just waiting for this branch to be brought up to date and deployed.

File: app/tests/unit/forms/form/controller.spec.js [hunk 2]
```diff
@@ -50,6 +52,21 @@ describe('form controller', () => {
     await controller.readFieldsForCSVExport(req, {}, jest.fn());
     expect(exportService.fieldsForCSVExport).toHaveBeenCalledTimes(1);
   });
+
+  it('should not continue with export if there are no submissions', async () => {
```
Comment (WalterMoar) [inline app/tests/unit/forms/form/controller.spec.js:56]: It's great to see new tests for the backend! It looks like the frontend changes don't have any test coverage, can those be added too? I assume we're still aiming for unit test coverage and the Cypress tests are focused on e2e?

File: app/frontend/tests/unit/views/file/DownloadTwo.spec.js
Comment (jasonchung1871) [inline app/frontend/tests/unit/views/file/DownloadTwo.spec.js:1]: Did something weird here. I'm unable to figure out exactly why all subsequent calls to the method `getFile` get skipped after the first test, so I needed to create a second file containing the next test. This should be added to the backlog as a bug that needs to be looked into as this could be affecting other component tests.

File: app/frontend/src/components/forms/ExportSubmissions.vue [hunk 3]
```diff
@@ -82,7 +82,9 @@ export default {
     },
     FILTER_HEADERS() {
       return this.versionSelected !== ''
-        ? this.formFields.map((f) => ({ name: f, value: f }))
+        ? this.formFields.length > 0
```
Comment (jasonchung1871) [inline app/frontend/src/components/forms/ExportSubmissions.vue:85]: This allows the formFields to be empty as sometimes when generating the headers for a submission, it can be empty because there are no submissions. In theory, it should contain the form fields from the form design but this is a separate issue.

File: app/frontend/src/views/file/Download.vue [hunk 4]
```diff
@@ -45,11 +45,7 @@ export default {
     async getFile(fileId) {
       this.showDownloadLink = false;
       await this.downloadFile(fileId);
-      if (
-        this.downloadedFile &&
-        this.downloadedFile.data &&
-        this.downloadedFile.headers
-      ) {
+      if (this.downloadedFile && this.downloadedFile.headers) {
```
Comment (jasonchung1871) [inline app/frontend/src/views/file/Download.vue:48]: This allows the download to begin if there is no data inside of the file to be downloaded, no submissions permitted. This way the user is actually notified that something happened instead of it just spinning infinitely.
